pkg/project/project_test.go

```diff
@@ -0,0 +1,63 @@
+package project
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/bouk/monkey"
+	"github.com/redhat-developer/odo/pkg/occlient"
+	"github.com/redhat-developer/odo/pkg/testingutil"
+	"k8s.io/apimachinery/pkg/runtime"
+	ktesting "k8s.io/client-go/testing"
+)
+
+func TestDelete(t *testing.T) {
+	tests := []struct {
+		name        string
+		wantErr     bool
+		projectName string
+	}{
+		{
+			name:        "Test project delete for multiple projects",
+			wantErr:     false,
+			projectName: "prj2",
+		},
+		{
+			name:        "Test only project delete",
+			wantErr:     false,
+			projectName: "prj1",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			// Fake the client with the appropriate arguments
+			client, fakeClientSet := occlient.FakeNew()
+
+			fakeClientSet.ProjClientset.PrependReactor("list", "projects", func(action ktesting.Action) (bool, runtime.Object, error) {
+				return true, testingutil.FakeProjects(), nil
+			})
+
+			fakeClientSet.ProjClientset.PrependReactor("delete", "projects", func(action ktesting.Action) (bool, runtime.Object, error) {
+				return true, nil, nil
+			})
+
+			var guard *monkey.PatchGuard
```

Comment:
This is new to me as well as our testing suite. Please comment on what's going on here with "monkey"

Reply:
Ack